test_regress/t/t_interface_param_dependency.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3
+# DESCRIPTION: Test interface parameter dependency resolution
+#
+# Copyright 2024 by Paul Swirhun. This program is free software; you
+# can redistribute it and/or modify it under the terms of either the GNU
+# Lesser General Public License Version 3 or the Perl Artistic License
+# Version 2.0.
+# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+
+import vltest_bootstrap
+
+test.scenarios("simulator_st")
+
+test.compile()
+
+test.execute(expect_filename=test.golden_filename)
+
```

Comment:
```suggestion
test.execute()

test.files_identical_sorted(test.run_log_filename, test.golden_filename, is_logfile=True)
```

As the initial order is not deterministic and might change and fail.  BTW generally I'd prefer use of `checkd to make the test self-checking without output comparison (as is easier to test on other simulators).